File: NextUC/app/ViewModelDelegate.swift
//
//  ViewModelDelegate.swift
//  NextUC
//
//  Created by Steve Chen on 3/5/16.
//  Copyright © 2016 Steve Chen. All rights reserved.
//

import Foundation

class ViewModelDelegate: CoreIViewModelDelegate {
    
    weak var mDelegate: CoreIViewModelDelegate?
    
    init(delegate: CoreIViewModelDelegate) {
        mDelegate = delegate
    }
    
    @objc func onDataChanged() {
        mDelegate?.onDataChanged()
    }
    
}

File: core/platform/android/java/delegate/ViewModelDelegate.java
package delegate;

import com.nextuc.IViewModelDelegate;

/**
 * Created by coa.ke on 7/28/16.
 */

public class ViewModelDelegate implements IViewModelDelegate {
    private IViewModelDelegate mViewModelDelegate;

    public ViewModelDelegate(IViewModelDelegate mViewModelDelegate) {
        this.mViewModelDelegate = mViewModelDelegate;
    }

    @Override
    public void onDataChanged() {
        mViewModelDelegate.onDataChanged();
    }
}
